distributed/s3.py

```diff
@@ -0,0 +1,31 @@
+import boto3
+from dask.imperative import Value
+from distributed import default_executor
+
+DEFAULT_PAGE_LENGTH = 1000
+
+
+def get_list_of_summary_objects(bucket_name, prefix='', delimiter='', page_size=DEFAULT_PAGE_LENGTH):
+    s3 = boto3.resource('s3')
+    return list(s3.Bucket(bucket_name).objects.filter(Prefix=prefix, Delimiter=delimiter).page_size(page_size))
```

Comment:
Specifying a delimiter returns keys that contain the same prefix and the first occurrence of the delimiter after the prefix grouped together. 

Here is some documentation on using prefix and delimiter together: http://docs.aws.amazon.com/AmazonS3/latest/dev/ListingKeysHierarchy.html


Sent from my iPhone

> On Jan 18, 2016, at 10:27 AM, Matthew Rocklin <notifications@github.com> wrote:
> 
> In distributed/s3.py:
> 
> > @@ -0,0 +1,31 @@
> > +import boto3
> > +from dask.imperative import Value
> > +from distributed import default_executor
> > +
> > +DEFAULT_PAGE_LENGTH = 1000
> > +
> > +
> > +def get_list_of_summary_objects(bucket_name, prefix='', delimiter='', page_size=DEFAULT_PAGE_LENGTH):
> > +    s3 = boto3.resource('s3')
> > +    return list(s3.Bucket(bucket_name).objects.filter(Prefix=prefix, Delimiter=delimiter).page_size(page_size))
> How does S3 handle delimiters?
> 
> —
> Reply to this email directly or view it on GitHub.
>